apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/profile/ProfileTaskExecutionContext.java

```diff
@@ -174,6 +174,9 @@ private boolean tryToAttemptProfiling(TracingContext tracingContext, ID traceSeg
         for (int slot = 0; slot < profilingSegmentSlots.length; slot++) {
             if (profilingSegmentSlots[slot] == null) {
                 profilingSegmentSlots[slot] = segmentContext;
+
+                // see https://www.javamex.com/tutorials/volatile_arrays.shtml, solution 2
```

Comment:
Don't put the link here. The article is just reference, we don't have the license to broadcast it.